File: backend/app/api/v1/schemas/my_work.py
"""`/my-work` — one response, five "waiting on me" queues.

Every field here is rendered, never a bare id: `WorkItem.title` is a NAME
(environment, release, incident — see the display-names rule in CLAUDE.md),
because a personal inbox of things the reader has never seen is exactly where
a `#42` fallback is least readable.

`QueueResult.failed` is NOT the same as an empty queue (§5 — see
`my_work_service.build`'s per-queue try/except). A dashboard that goes blank
because one worklist query is unhappy is worse than one showing four of five
and saying so; the frontend renders `failed` and an empty `count`/`items`
differently for exactly that reason.
"""
from datetime import datetime

from pydantic import BaseModel


class WorkItem(BaseModel):
    id: int
    title: str            # a NAME, never "#42" — see the display-names rule
    subtitle: str | None = None
    url: str               # the detail route this row opens
    due: datetime | None = None


class QueueResult(BaseModel):
    count: int
    items: list[WorkItem]
    overdue: int | None = None   # pir_actions only
    failed: bool = False         # true => the queue could not be computed


class MyWorkResponse(BaseModel):
    as_of: datetime
    queues: dict[str, QueueResult]   # keys: environment_requests, contentions,
                                      # decommissions, pir_actions, incidents

File: frontend/src/types/myWork.ts
// GET /me/work — Phase "IA" Task 4/5: the five "waiting on me" queues a
// personal inbox composes under one clock. Mirrors
// backend/app/api/v1/schemas/my_work.py exactly.
//
// `title` is ALWAYS a name, never a bare id — the backend never sends a
// `#42` fallback (see CLAUDE.md's display-names rule), so nothing here needs
// to resolve `id` against anything else to render a row.
export interface WorkItem {
  id: number;
  title: string;
  subtitle?: string | null;
  /** The detail route this row opens. */
  url: string;
  /** ISO datetime, or absent when the item carries no deadline. */
  due?: string | null;
}

/**
 * `failed` is NOT the same value as an empty queue (§5). A queue whose
 * underlying worklist query blew up comes back as `{ count: 0, items: [],
 * failed: true }` — indistinguishable from an empty one by `count`/`items`
 * alone, which is exactly why every consumer must branch on `failed` FIRST.
 */
export interface QueueResult {
  count: number;
  items: WorkItem[];
  /** `pir_actions` only. */
  overdue?: number;
  failed: boolean;
}

export type MyWorkQueueKey =
  | 'environment_requests'
  | 'contentions'
  | 'decommissions'
  | 'pir_actions'
  | 'incidents';

export interface MyWorkResponse {
  as_of: string;
  queues: Record<MyWorkQueueKey, QueueResult>;
}
